src/main/java/com/ericsson/ei/subscription/SubscriptionRepeatDbHandler.java

```diff
@@ -58,178 +59,220 @@
+
+	    insertNewMatchedAggregationToDatabase(subscriptionId, requirementId,
+		    aggrObjId);
+	}
     }
 
-    public boolean checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(String subscriptionId, int requirementId,
-                                                                         String aggrObjId) {
-
-        LOGGER.debug("Checking if AggrObjId: " + aggrObjId + " exist in SubscriptionId: " + subscriptionId
-                + " AggrId matched list.");
-        String subscriptionQuery = "{\"subscriptionId\" : \"" + subscriptionId + "\"}";
-        List<String> objArray = mongoDbHandler.find(dataBaseName, collectionName, subscriptionQuery);
-        if (objArray != null && !objArray.isEmpty()) {
-
-            LOGGER.debug("Making AggrObjId checks on SubscriptionId document: " + objArray.get(0));
-            try {
-                JsonNode jNode = mapper.readTree(objArray.get(0));
-                if (jNode.get("subscriptionId").asText().trim().equals(subscriptionId)) {
-                    LOGGER.debug("SubscriptionId \"" + subscriptionId +
-                            "\" , exist in document. Checking if AggrObjId has matched earlier.");
-
-                    LOGGER.debug("Subscription requirementId: " + requirementId + " and Requirements content:\n"
-                            + jNode.get("requirements").get(new Integer(requirementId).toString()));
-
-                    boolean triggered = checkRequirementIdTriggered(jNode, requirementId, aggrObjId);
-
-                    if (!triggered) {
-                        LOGGER.debug("RequirementId: " + requirementId + " and SubscriptionId: " + subscriptionId
-                                + "\nhas not matched any AggregatedObject.");
-                    }
-                    return triggered;
-                }
-            } catch (Exception e) {
-                LOGGER.error(e.getMessage(), e);
-                e.printStackTrace();
-            }
-        }
-        LOGGER.debug("AggrObjId: " + aggrObjId + "not found for SubscriptionId: " + subscriptionId
-                + " in SubscriptionRepeatFlagHandlerDb -> Returning FALSE.");
-        return false;
+    public boolean checkIfAggrObjIdExistInSubscriptionAggrIdsMatchedList(
+	    String subscriptionId, int requirementId, String aggrObjId) {
+
+	LOGGER.debug(
+		"Checking if AggrObjId: {} exist in SubscriptionId: {} AggrId matched list.",
+		aggrObjId, subscriptionId);
+	String subscriptionQuery = "{\"subscriptionId\" : \"" + subscriptionId
+		+ "\"}";
+	List<String> objArray = mongoDbHandler.find(dataBaseName,
+		collectionName, subscriptionQuery);
+	if (objArray != null && !objArray.isEmpty()) {
+
+	    LOGGER.debug(
+		    "Making AggrObjId checks on SubscriptionId document: {}",
+		    objArray.get(0));
+	    try {
+		JsonNode jNode = mapper.readTree(objArray.get(0));
+		if (jNode.get("subscriptionId").asText().trim()
```

Comment:
Please move jNode.get("subscriptionId").asText().trim()....... to a variable.

Reply:
Done.